### PR TITLE
Check for deprecated `typing.Set` in `check_new_syntax.py`

### PR DESCRIPTION
**Pull request by AlexWaygood**
This PR proposes checking for use of the deprecated alias `typing.Set` in `check_new_syntax.py`. It's a little tricky, as `typing.Set` and `collections.abc.Set` are not the same class.

### CODE REVIEW

File: tests/check_new_syntax.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import re
 import sys
 from itertools import chain
 from pathlib import Path
 
+SINGLE_LINE_COLLECTIONS_SET_IMPORT_REGEX = re.compile(r"from\s+collections\.abc\s+import\s+(\w+\s*,\s*)*Set")
+# This one won't match unless the code's been formatted with black.
+MULTI_LINE_COLLECTIONS_SET_IMPORT_REGEX = re.compile(
+    r"from\s+collections\.abc\s+import\s*\((\n    \w+,\s*)*\n    Set(\s*,\s*\n    \w+)*\s*,?\s*\)"
+)
```
Comment (Akuli) [inline tests/check_new_syntax.py:13]: It would be better to add a `visit_Import()` method to `OldSyntaxFinder`.

Comment (AlexWaygood) [reply tests/check_new_syntax.py:13]: Fair, I'm more confident using regular expressions which is why I went that route, but I'll give it a go.

Comment (Akuli) [reply tests/check_new_syntax.py:13]: Actually, I think the correct name is `visit_ImportFrom`.

```
>>> import ast
>>> ast.parse('from collections.abc import Set')
<_ast.Module object at 0x7f8cae6ae630>
>>> print(ast.dump(_))
Module(body=[ImportFrom(module='collections.abc', names=[alias(name='Set', asname=None)], level=0)])
>>> 
```

Comment (AlexWaygood) [reply tests/check_new_syntax.py]: How's the new version using AST? (You're... right, using AST is much easier!)

**Comment by Akuli** [comment]
Thanks! Can you replace built-in `set` with `from collections.abc import Set` in some file, then push that commit, and then revert and push? This way anyone looking at the pull request can be confident that it errors as intended.

**Comment by Akuli** [comment]
Try again, pushing one commit at a time. If you push multiple commits at once, only the last commit gets a CI run.

**Comment by AlexWaygood** [comment]
> **Edit:** nevermind, the second commit wasn't a revertion :)

Nope, it was a correction of my coding mistake ;) Good job you made me check!

**Review by Akuli — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks @Akuli!

**Pull request merged**